src/pip/_internal/resolution/resolvelib/requirements.py

```diff
@@ -16,9 +17,10 @@
 
 
 class ExplicitRequirement(Requirement):
-    def __init__(self, candidate):
-        # type: (Candidate) -> None
+    def __init__(self, candidate, factory):
+        # type: (Candidate, Factory) -> None
         self.candidate = candidate
+        self._factory = factory
```

Comment:
This attribute is not used anywhere anymore (since the constraints are now passed into `find_matches()` as an argument).

Reply:
Good catch! I'll fix that.